### PR TITLE
[10.0][ADD] cagnotte_sale: Add module to manage cagnotte on sale

### PR DESCRIPTION
**Pull request by rousseldenis**
Based on https://github.com/acsone/acsone-addons/pull/147

### CODE REVIEW

File: cagnotte_sale/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        return self.amount_total
+
+    @api.multi
+    def unset_cagnotte(self):
+        self.mapped('order_line').filtered('account_cagnotte_id').unlink()
+
+    @api.multi
+    def apply_cagnotte(self, cagnotte):
+        for sale in self:
+            if cagnotte.partner_id != sale.partner_id:
```
Comment (Cedric-Pigeon) [inline cagnotte_sale/models/sale_order.py]: This won't work. The partner_id on the cagnotte is not mandatory.  When a customer buys a giftcard for someone else, no partner is set.

File: cagnotte_sale/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                raise UserError(_('The cagnotte you try to use is not yours!'))
+            # We check that the sale order value is > 0
+            if cagnotte.solde_cagnotte > 0.0 and self.amount_total > 0.0:
```
Comment (Cedric-Pigeon) [inline cagnotte_sale/models/sale_order.py]: would use `float_compare`

**Review by Cedric-Pigeon — CHANGES_REQUESTED** [review]

File: cagnotte_sale/models/sale_order.py (hunk 1 above)
Comment (rousseldenis) [reply cagnotte_sale/models/sale_order.py]: Ok, I had a doubt on that

File: cagnotte_sale/models/sale_order.py (hunk 2 above)
Comment (rousseldenis) [reply cagnotte_sale/models/sale_order.py]: I'm not sure. I need to see in code how Odoo is treating Monetary fields (they are not pure float, so no decimal precision)

**Pull request closed without merging**